server/confluence_config_test.go

```diff
@@ -0,0 +1,117 @@
+package main
+
+import (
+	"bytes"
+	"fmt"
+	"net/http"
+	"net/http/httptest"
+	"reflect"
+	"testing"
+
+	"bou.ke/monkey"
+	"github.com/mattermost/mattermost-plugin-confluence/server/config"
+	"github.com/mattermost/mattermost-server/v6/model"
+	"github.com/mattermost/mattermost-server/v6/plugin"
+	"github.com/mattermost/mattermost-server/v6/plugin/plugintest/mock"
+	"github.com/stretchr/testify/assert"
+)
+
+func TestHandleConfluenceConfig(t *testing.T) {
+	tests := map[string]struct {
+		method         string
+		statusCode     int
+		body           string
+		userID         string
+		channelID      string
+		patchFuncCalls func()
+	}{
+		"success": {
+			method:     http.MethodPost,
+			statusCode: http.StatusOK,
+			body: `{
+				"type": "dialog_submission",
+				"callback_id": "callbackID",
+				"state": "", 
+				"submission": {
+					"Client ID": "mock-ClientID",
+					"Client Secret": "mock-ClientSecret",
+					"Server URL": "https://test.com"
+				},
+				"cancelled": false
+			}`,
+			userID:    "iu73atknztnctef8b8ey9gm6zc",
+			channelID: "tgniw3kmrjd93qns11cboditme",
+			patchFuncCalls: func() {
+				monkey.PatchInstanceMethod(reflect.TypeOf(&Plugin{}), "GetConfigKeyList", func(_ *Plugin) ([]string, error) {
+					return []string{
+						"https://test.com",
+					}, nil
+				})
+			},
+		},
+		"wrong api method": {
```

Comment:
```suggestion
		"wrong API method": {
```
Fix this everywhere.